src/classes/COFF_Affiliation_TDTM.cls

```diff
@@ -268,7 +276,8 @@ public class COFF_Affiliation_TDTM extends TDTM_Runnable {
     * @description Sets each Affiliation pointing to the new Faculty member that is teaching the Course Offering.
     ********************************************************************************************************/
     private List<SObject> getAfflsToUpdate(List<ID> outdatedFacultyIDs, List<ID> updatedFacultyIDs, Map<Id, Course_Offering__c> mapNewFacultyIdToNewCourseOffering) {
-        List<Affiliation__c> afflsWithContact = [select Contact__c from Affiliation__c where Contact__r.ID in :outdatedFacultyIDs and Role__c = 'Faculty'];
+        List<Affiliation__c> afflsWithContact = [SELECT Contact__c FROM Affiliation__c WHERE Contact__c IN :outdatedFacultyIDs AND Role__c = 'Faculty'];
```

Comment:
It looks like you're simply fixing what was already there so this is not you, BUT -- this query will break if a customer decides to encrypt the Role__c field I believe. Since encrypted fields can't be used in SOQL filters and ORDER BY statements. What really sucks about that is, it would force us to query on all of those whose Contact Id is in that list, and then loop over to filter out non Faculty values.

It's the same in the unit tests below. We cannot have queries there with the strings if those fields get encrypted. It's an extra step for us, but we just announced a release or two ago that we were now Shield compliant.

However, since it looks like you were simply fixing something we had there already (3 years ago it appears) and our scans didn't seem to find it when we were encrypting fields, I think we can leave it for now. I might just test encrypting that field when I pull this down to see if it freaks out. (Maybe its find if its a picklist value?)